R/rowClust.R

```diff
@@ -0,0 +1,41 @@
+# REQUIRED PACKAGES: "fastcluster", "stats", "coop"
+rowClust <- function(tse,
+                     abund_values = "counts",
+                     metric = "spearman", # c("pearson", "kendall", "spearman" || 
+                                          #    "cosine" ||
+                                          #    "euclidean", "maximum", "manhattan", "canberra", "binary", "minkowski")
+                     method = "ward.D2", # c("single", "complete", "average", "mcquitty", "ward.D", "ward.D2", "centroid","median")
+                     name = "clust_id",
+                     k = NULL, # number of clusters
+                     h = NULL, # height of cut-off
+                     ... # arguments passed to stats::dist (mainly for the p arg of "minkowski")
+                     ) {
+  # Getting Assay data
+  if(abund_values %in% assayNames(tse))
+    x <- assay(tse, abund_values)
+  else
+    stop(paste0("Assay ",abund_values," not Found!"))
```

Comment:
Add `.call = FALSE`. This removes function name from error message / makes the message more clear (At least if this error would be inside internal function)

Like this:
```
stop("'rank' must be an single character value.",
                     call. = FALSE)
```